### PR TITLE
Enable sriovNet and ENA for wrapped image/instance

### PR DESCRIPTION
**Pull request by krishnanrs**
Enable sriovNetSupport and ENA for a wrapped image or instance, if
necessary

### CODE REVIEW

File: brkt_cli/aws/aws_service.py [hunk 1]
```diff
@@ -1194,13 +1194,29 @@ def get_code_and_message(client_error):
...
+    :param instance an ec2.Instance or ec2.Image object
     """
-    if not hasattr(instance, 'ena_support'):
+    if not hasattr(resource, 'ena_support'):
         return False
-    return bool(instance.ena_support)
+    return bool(resource.ena_support)
+
+
+def enable_sriovNetSupport(aws_svc, instance):
```
Comment (bburtin) [inline brkt_cli/aws/aws_service.py]: This should probably be `enable_sriov_net_support`, to follow convention.

File: brkt_cli/aws/aws_service.py [hunk 2]
```diff
@@ -1194,13 +1194,29 @@ def get_code_and_message(client_error):
...
+    if instance.sriov_net_support != "simple":
+        log.info('Enabling sriovNetSupport for %s', instance.id)
+        try:
+            aws_svc.modify_instance_attribute(
+                instance.id,
+                "sriovNetSupport",
+                "simple")
+            log.info('sriovNetSupport enabled successfully')
+        except ClientError as e:
+            log.warn('Unable to enable sriovNetSupport for instance '
```
Comment (bburtin) [inline brkt_cli/aws/aws_service.py]: Just to confirm: if modifying the instance attribute fails, the user will still have a working instance?  If this isn't the case, we should probably abort.

File: brkt_cli/aws/wrap_image.py [hunk 3]
```diff
@@ -241,6 +242,25 @@ def wrap_instance(aws_svc, instance_id, metavisor_ami, instance_config=None):
             delete_on_termination=True
         )
 
+        # Enable sriovNetSupport
```
Comment (bburtin) [inline brkt_cli/aws/wrap_image.py]: Is there enough overlap here that we can turn this into two functions that are called from the encrypt, update, and wrap code?

File: brkt_cli/aws/aws_service.py (hunk 2 above)
Comment (krishnanrs) [reply brkt_cli/aws/aws_service.py]: That is correct. The instance will continue to work, just will not be able to utilize enhanced networking.

File: brkt_cli/aws/wrap_image.py (hunk 3 above)
Comment (krishnanrs) [reply brkt_cli/aws/wrap_image.py]: I didn't understand this comment. sriovNetSupport is enabled only in encrypt and wrap-instance/image command. For this I moved it into a common function which is being invoked from these two places. Am I missing something? Please clarify.

Comment (bburtin) [reply brkt_cli/aws/wrap_image.py]: Oh... ok.  I remembered something about sriovNet from before, and assumed it was in the encrypt path.  Sorry if this is a false alarm.

**Pull request merged**